lib/wallaby/element.ex

```diff
@@ -284,11 +237,34 @@ defmodule Wallaby.Element do
   @spec location(t) :: t
 
   def location(%__MODULE__{driver: driver} = element) do
-    case driver.element_location(element) do
-      {:ok, value} ->
-        value
+    element
+    |> driver.element_location()
+    |> handle_value_result()
+  end
+
+  defp handle_action_result(result, element) do
+    case result do
+      {:ok, _} -> element
+      {:error, error} -> raise_error(error)
     end
   end
+
+  defp handle_value_result(result) do
+    case result do
+      {:ok, value} -> value
+      {:error, error} -> raise_error(error)
+    end
+  end
+
+  defp handle_boolean_result(result) do
+    case result do
+      {:ok, true} -> true
+      {:ok, false} -> false
+      {:error, error} -> raise_error(error)
+    end
+  end
+
+  defp raise_error(:stale_reference), do: raise(StaleReferenceError)
```

Comment:
WDYT about adding a clause that raises a runtime error with the argument? Might be better than a FunctionClauseError.

Reply:
My thinking was that this would surface errors we're not properly handling, but I'm OK with your suggestion (which would still surface errors we're not properly handling...) :+1: